src/main/java/com/suse/saltstack/netapi/parser/SaltStackParser.java

```diff
@@ -1,32 +1,39 @@
 package com.suse.saltstack.netapi.parser;
 
 import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
 import com.suse.saltstack.netapi.exception.SaltStackParsingException;
+import com.suse.saltstack.netapi.results.*;
 
 import java.io.BufferedReader;
 import java.io.InputStream;
 import java.io.InputStreamReader;
 import java.io.Reader;
-import java.lang.reflect.Type;
+import java.util.List;
 
-/**
- * ABS for result parsers. All result parsers need to inherit from this class. Generic methods
- * that apply to multiple results can be added to this class.
- */
-public abstract class SaltStackParser<T> {
-    public abstract T parse(InputStream inputStream) throws SaltStackParsingException;
+
+public class SaltStackParser<T> {
+
+    public static final SaltStackParser<SaltStackResult<String>> STRING = new SaltStackParser<>(new TypeToken<SaltStackResult<String>>(){});
+    public static final SaltStackParser<SaltStackResult<List<SaltStackToken>>> TOKEN = new SaltStackParser<>(new TypeToken<SaltStackResult<List<SaltStackToken>>>(){});
+    public static final SaltStackParser<SaltStackResult<List<SaltStackJob>>> JOB = new SaltStackParser<>(new TypeToken<SaltStackResult<List<SaltStackJob>>>(){});
```

Comment:
Yes, please wrap long lines. It's not (yet) written down somewhere in this repo, but I tend to like a maximum line length of 92 characters. Please apply this to other long lines as well.